pandas/core/arrays/timedeltas.py

```diff
@@ -214,6 +214,19 @@ def _simple_new(
 
     @classmethod
     def _from_sequence(
+        cls, data, dtype=TD64NS_DTYPE, copy: bool = False
+    ) -> "TimedeltaArray":
+        if dtype:
+            _validate_td64_dtype(dtype)
+
+        data, inferred_freq = sequence_to_td64ns(data, copy=copy, unit=None)
+        freq, _ = dtl.validate_inferred_freq(None, inferred_freq, False)
+
+        result = cls._simple_new(data, freq=freq)
+        return result
+
+    @classmethod
+    def _from_sequence_not_strict(
```

Comment:
umm, have we standarized on this naming? or even the concept of 2 from_sequence?
cc @jorisvandenbossche @TomAugspurger 

IIRC we had a need for a `coerce=` keyword that we nixed for StringArray

Reply:
xref https://github.com/pandas-dev/pandas/pull/36718 as well

Reply:
yah my best guess is we're gonna end up dropping this method altogether and having it only reached via the DTI/TDI constructors

Reply:
ok as a temporary work-around prob ok.

Reply:
yah, there's also more work to do in tightening what dtypes _from_sequence accepts, plenty left to do